state/txn.go

```diff
@@ -338,10 +338,15 @@ func (txn *Txn) GetState(addr types.Address, key types.Hash) types.Hash {
 // Nonce
 
 // IncrNonce increases the nonce of the address
-func (txn *Txn) IncrNonce(addr types.Address) {
+func (txn *Txn) IncrNonce(addr types.Address) (err error) {
 	txn.upsertAccount(addr, true, func(object *StateObject) {
+		if object.Account.Nonce+1 < object.Account.Nonce {
+			err = errors.New("nonce uint64 overflow")
+		}
 		object.Account.Nonce++
 	})
+
+	return
 }
```

Comment:
If I'm not mistaken, this will increment nonce even if there is an error

```suggestion
func (txn *Txn) IncrNonce(addr types.Address) error {
	txn.upsertAccount(addr, true, func(object *StateObject) {
		if object.Account.Nonce+1 < object.Account.Nonce {
			return errors.New("nonce uint64 overflow")
		}
		object.Account.Nonce++
	})

	return nil
}
```